gitautosync/pull_from_remote.py

```diff
@@ -0,0 +1,129 @@
+import os
+import re
+import subprocess
+from .util import logger
+
+DELETED_FILE_REGEX = re.compile(
+    r"deleted:\s+"  # Look for deleted: + any amount of whitespace...
+    r"([^\n\r]+)"   # and match the filename afterward.
+)
+
+
+def pull_from_remote(git_url, branch_name, repo_dir):
+    """
+    Pull selected repo from a remote git repository,
+    while preserving user changes
+    """
+    assert git_url and branch_name
+
+    logger.info('Starting pull.')
```

Comment:
Log lines should be terse. This should instead just be 'Pulling into {} from {}, branch {}'. Multi line logs make grepping much harder